pkg/languages/defaults/python.go

```diff
@@ -16,8 +18,32 @@ func (p PythonExtractor) ReadDefaults(r reporeader.RepoReader) (map[string]strin
 	if err != nil {
 		return nil, fmt.Errorf("error finding python files: %v", err)
 	}
-	if len(files) > 0 {
-		extractedValues["ENTRYPOINT"] = files[0]
+
+	entryPointPattern := `if\s*__name__\s*==\s*["']__main__["']`
+	compiledPattern, err := regexp.Compile(entryPointPattern)
+	if err != nil {
+		return nil, fmt.Errorf("error compiling regex pattern: %v", err)
+	}
+
+	for _, filePath := range files {
```

Comment:
adding some comments for the reasoning for each of the successive passes here would make this even more friendly for future developers who may add other languages' extractors based on this one :)

Reply:
Added a few comments. Lmk if it looks good.